dart/lib/src/sentry_client.dart

```diff
@@ -177,41 +177,49 @@ class SentryClient {
     final isolateId = isolateName?.hashCode;
 
     if (event.throwableMechanism != null) {
-      var sentryException = _exceptionFactory.getSentryException(
-        event.throwableMechanism,
-        stackTrace: stackTrace,
-      );
+      final throwable = event.throwable;
+      final extractedExceptions = _options.extractor.flatten(throwable, stackTrace);
+
+      var sentryExceptions = <SentryException>[];
+      var sentryThreads = <SentryThread>[];
 
-      if (_options.platformChecker.isWeb) {
-        return event.copyWith(
-          exceptions: [
-            ...?event.exceptions,
-            sentryException,
-          ],
+      for (final extractedException in extractedExceptions) {
+        var sentryException = _exceptionFactory.getSentryException(
+          extractedException.exception,
+          stackTrace: extractedException.stackTrace,
         );
-      }
 
-      SentryThread? thread;
+        SentryThread? sentryThread;
 
-      if (isolateName != null && _options.attachThreads) {
-        sentryException = sentryException.copyWith(threadId: isolateId);
-        thread = SentryThread(
-          id: isolateId,
-          name: isolateName,
-          crashed: true,
-          current: true,
-        );
+        if (!_options.platformChecker.isWeb &&
+            isolateName != null &&
+            _options.attachThreads) {
+          sentryException = sentryException.copyWith(threadId: isolateId);
+          sentryThread = SentryThread(
+            id: isolateId,
+            name: isolateName,
+            crashed: true,
+            current: true,
+          );
+        }
+
+        sentryExceptions.add(sentryException);
+        if (sentryThread != null) {
+          sentryThreads.add(sentryThread);
+        }
       }
 
       return event.copyWith(
-        exceptions: [...?event.exceptions, sentryException],
+        exceptions: [...?event.exceptions, ...sentryExceptions],
         threads: [
           ...?event.threads,
-          if (thread != null) thread,
+          ...sentryThreads,
         ],
       );
     }
 
+    // TODO How to handle boxed cause with stacktrace only?
```

Comment:
Can you elaborate?

Reply:
I guess he refers to https://github.com/getsentry/sentry-dart/pull/1198#discussion_r1052426360

FWIW, `Sentry.captureException()` doesn't report anything if the exception is null.

Reply:
So if the exception is null, we don't do anything.
If there's an exception/cause, and no stack trace, there is a few approaches, fall back to `Stacktrace.current`, don't add a stack trace for the cause, or use the stack trace of the parent.

Reply:
Yeah, was refering to "An exception has just an inner stacktrace". Dunno if we should do anything special in such a case?

Reply:
If the exception is null, it bails out early, nothing to do there.